src/Adapter/Order/Repository/OrderRepository.php

```diff
@@ -71,4 +77,37 @@ public function get(OrderId $orderId): Order
 
         return $order;
     }
+
+    /**
+     * @param OrderDetailId $detailId
+     * @param LanguageId $languageId
+     *
+     * @return OrderDetailCustomizations|null
+     *
+     * @throws PrestaShopException
+     * @throws \PrestaShopDatabaseException
```

Comment:
shouldn't be thrown.
Repository should wrap the old code and don't let the legacy exceptions to bubble up, but instead throw CoreException (check examples in abstract repository)